annotations/views/rest_views.py

```diff
@@ -493,8 +501,71 @@ def _relabel(datum):
                 'identifier': 'uri'
             }
 
-            return {_fields.get(k, k): v for k, v in datum.iteritems()}
-        return Response({'results': map(_relabel, [c.data for c in concepts])})
+            return {_fields.get(k, k): v for k, v in datum.iteritems() }
+        results = map(_relabel, [c.data for c in concepts])
+        
+        for result in results:
+            concepts = []
+            if result["identities"]: # if identities exist append the first identitiy to the list so that we can filter out other identities against it
+                concepts = result["identities"][0]["concepts"]
+            conceptsSet = Set(concepts)
+            for ident in result["identities"]: # go through the identities in each result
+                identSet = Set(ident['concepts'])
+                diff = identSet.difference(conceptsSet)
+                conceptsSet.update(diff)
+            result["identities"] = list(conceptsSet) # replace the identities list
+            if result["identities"]:
+                concepts = result["identities"]
```

Comment:
instead of this, you should be able to say
```
concepts = results["identities"] = list(conceptSet)
```

Reply:
and then keep using ```concepts``` in the if (and if condition)

Reply:
changed